airflow/executors/base_executor.py

```diff
@@ -216,6 +215,8 @@ def heartbeat(self) -> None:
         self.log.debug("%s in queue", num_queued_tasks)
         self.log.debug("%s open slots", open_slots)
 
+        from airflow.stats import Stats
```

Comment:
Does this really affact import time much? (Also the `exceptions` one below.)

From what I can tell these modules are very barebone and should only cost a fraction to import? And the pure import time save seems marginal at best since they need to be imported for pretty much anything anywhere for the executor to be actually useful.

Reply:
Thanks for having a look TP!

> From what I can tell these modules are very barebone and should only cost a fraction to import?

When I tested these two modules individually they were taking 600-800ms to load (vs cheap modules taking an order of magnitude less such as 50-90ms).

> And the pure import time save seems marginal at best since they need to be imported for pretty much anything anywhere for the executor to be actually useful.

In cases like the CLI usecase, we're not using the executor in that way (i.e. instantiating it) so any savings on import can be quite impactful actually.

Reply:
I think those measurments for stats come from running it without "from airflow.configuration import conf".  This is one of the most complex imports, and is about the only thing that seem to have measureable impect when importing stats module. 

If you look up the file it is already imported  up there, conf is imported up there, so I think @uranusjr is right about this one, Same with the "Exceptions"

Also why the measurement can be biased (for exceptions case) is that when we are importing any airflow package, we  also import `airflow.__init__` and this one is also rather slow-ish  with configuration and settings import. I tihnk this might have (If I am not mistaken) significant impct on the times measured for the imports done in isolation.

My tests for one have shown that once `airflow.configuration.conf` is imported, stats and exceptions import instanteneously.

Reply:
Thanks for the details @potiuk!

I've verified that stats is trivially fast when conf is already imported. I do actually have future plans to move conf imports closer to where they are used to speed up loading times further, and in that case it may be beneficial to move stat as well (but many things use conf so it may be difficult to remove it entirely from the whole imported namespace, we shall see).

If having Stats imported where I've moved it in this PR is unacceptable I can update the change :+1:

Reply:
Updated the PR to undo this portion :+1: